recipes/marisa/all/patches/0001-add-cmake.patch

```diff
@@ -0,0 +1,126 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+new file mode 100644
+index 0000000..1e1bda6
+--- /dev/null
++++ b/CMakeLists.txt
+@@ -0,0 +1,120 @@
++cmake_minimum_required(VERSION 3.1)
```

Comment:
Could you show me how to do that? Any ref.?

Reply:
Add as you did for other files in this PR:

git add CMakeLists.txt
git commit -s -m "Add Cmake file for marisa"
git push origin marisa/0.2.6

Reply:
Why not use the upstream's build system? https://github.com/s-yata/marisa-trie#build-instructions we have some docs on this https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#autotools

Also there's a year old PR https://github.com/s-yata/marisa-trie/pull/43 so usually I would suggestion making a PR to share your implementation but they are not very active

Reply:
Yes, @prince-chrismc approach is the better way.

Reply:
> Why not use the upstream's build system? https://github.com/s-yata/marisa-trie#build-instructions we have some docs on this https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#autotools
> 
> Also there's a year old PR [s-yata/marisa-trie#43](https://github.com/s-yata/marisa-trie/pull/43) so usually I would suggestion making a PR to share your implementation but they are not very active

The reason why I don't use upstream build system is:
- I want to add build support for Windows MSVC
- Since I keep facing issues like [this](https://github.com/conan-io/conan-center-index/issues/11965) when I tried to create recipes for `C++ Project` which use `autotools` as build system. #5863 has the same issue.
--> So I tried to use CMake system for this project.

Reply:
In that case the "Accepted" approach is something like https://github.com/conan-io/conan-center-index/blob/c4a7e6a366a1e600167da37f40be8cab13357358/recipes/giflib/5.2.x/CMakeLists.txt

There's some points in this CMake, like shared build, which are not required (CMake natively supports that option internal so you dont need to declare it) also there should not be any tests